src/rust/engine/fs/src/directory.rs

```diff
@@ -60,12 +63,55 @@ impl Debug for DirectoryDigest {
 }
 
 impl DirectoryDigest {
+  /// Construct a DirectoryDigest from a Digest and DigestTrie (by asserting that the Digest
+  /// identifies the DigestTrie).
+  pub fn new(digest: Digest, tree: DigestTrie) -> Self {
+    if cfg!(debug_assertions) {
+      assert!(digest == tree.compute_root_digest());
+    }
+    Self {
+      digest,
+      tree: Some(tree),
+    }
+  }
+
   /// Creates a DirectoryDigest which asserts that the given Digest represents a Directory structure
   /// which is persisted in a Store.
-  pub fn new(digest: Digest) -> Self {
+  ///
+  /// Use of this method should be rare: code should prefer to pass around a `DirectoryDigest` rather
+  /// than to create one from a `Digest` (as the latter requires loading the content from disk).
+  ///
+  /// TODO: If a callsite needs to create a `DirectoryDigest` as a convenience (i.e. in a location
+  /// where its signature could be changed to accept a `DirectoryDigest` instead of constructing
+  /// one) during the porting effort of #13112, it should use `todo_from_digest` rather than
+  /// `from_persisted_digest`.
+  pub fn from_persisted_digest(digest: Digest) -> Self {
+    Self { digest, tree: None }
+  }
+
+  /// Marks a callsite that is creating a `DirectoryDigest` as a temporary convenience, rather than
+  /// accepting a `DirectoryDigest` in its signature. All usages of this method should be removed
+  /// before closing #13112.
+  pub fn todo_from_digest(digest: Digest) -> Self {
     Self { digest, tree: None }
   }
 
+  /// Discards the `DigestTrie` held by this `DirectoryDigest`.
```

Comment:
This function doesn't mutate anything, is this description accurate?

Reply:
Probably a bit of a stretch. It's more accurate for `todo_as_digest`, where the intent is to discard the tree for temporary convenience.